sumologic/resource_sumologic_cse_match_list.go

```diff
@@ -189,29 +189,47 @@ func resourceSumologicCSEMatchListCreate(d *schema.ResourceData, meta interface{
 			}
 		}
 
-		createStateConf := &resource.StateChangeConf{
-			Target: []string{
-				fmt.Sprint(len(items)),
-			},
-			Refresh: func() (interface{}, string, error) {
-				resp, err := c.GetCSEMatchListItemsInMatchList(d.Id())
-				if err != nil {
-					log.Printf("[ERROR] CSE Match List items not found when looking by match list id %s, err: %v", d.Id(), err)
-					return 0, "", err
-				}
-				return resp, fmt.Sprint(resp.Total), nil
-			},
-			Timeout:                   d.Timeout(schema.TimeoutCreate),
-			Delay:                     10 * time.Second,
-			MinTimeout:                5 * time.Second,
-			ContinuousTargetOccurence: 1,
+		var createStateConf *resource.StateChangeConf
+		if len(items) <= 10000 {
```

Comment:
Why use a different endpoint for less than 10K items? Is it better/faster somehow? I'd prefer the code to use one endpoint, if feasible.